src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceDriveIfAvailable.scala

```diff
@@ -6,16 +6,18 @@ import beam.router.Modes.BeamMode.CAR
 import beam.router.RoutingModel.EmbodiedBeamTrip
 import beam.sim.BeamServices
 
+import scala.collection.mutable.ArrayBuffer
+
 /**
   * BEAM
   */
 class ModeChoiceDriveIfAvailable(val beamServices: BeamServices) extends ModeChoiceCalculator {
 
   def apply(alternatives: Seq[EmbodiedBeamTrip]): Option[EmbodiedBeamTrip] = {
-    var containsDriveAlt: Vector[Int] = Vector[Int]()
+    val containsDriveAlt: ArrayBuffer[Int] = ArrayBuffer[Int]()
     alternatives.zipWithIndex.foreach { alt =>
```

Comment:
Fix this with:
```
alternatives.zipWithIndex.filter { case (beamTrip, idx) => beamTrip. tripClassifier == CAR }.map { case (beamTrip, idx) => idx}

```

Reply:
Even better would be:

`val containsDriveAlt = alternatives.zipWithIndex.collect{ case(trip, idx) if(trip.tripClassifier == CAR) => idx}`